lib/apiservers/portlayer/swagger.json

```diff
@@ -175,8 +176,166 @@
+				      "tags": [
+					        "storage"
+				      ],
+				      "consumes": [
+					        "application/json"
+				      ],
+				      "produces": [
+					        "application/x-tar"
+				      ],
+				      "operationId": "ExportArchive",
+				      "parameters": [
+						{
+							"name": "store",
+							"in":"query",
+							"required":true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name": "ancestorStore",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"deviceID",
+							"in":"query",
+							"required": true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"ancestor",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"data",
+							"in":"query",
+							"required": true,
+							"type" : "boolean",
+							"x-nullable": false						
+						},
+						{
+							"name":"filterSpec",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": true
+						}
+					],
+					"responses": {
+						"200": {
+							"description": "OK",
+							"schema": {
+								"format": "binary"
+							}
+						},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"423": {
+						"description": "Device or resource is locked",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"500": {
+						"description": "failed to export tar archive from target"
+					},
+					"default": {
+						"description": "error",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					}
+				}
+          },
+      "post": {
+          "description": "Imports a tar stream to a device at the target path",
+				  "summary": "Import tar stream to supplied target",
+				  "tags": [
+					        "storage"
+				  ],
+				  "consumes": [
+					    "application/x-tar"
+				  ],
+				  "operationId": "ImportArchive",
+				  "parameters": [
+					{
+						"name": "store",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"deviceID",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"filterSpec",
+						"in":"query",
+						"required": false,
+						"type" : "string",
+						"x-nullable": true
+					},
+					{
+						"name": "Archive",
+						"required": true,
+						"in": "body",
+						"schema": {
+							"type":"string",
+							"format":"binary"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
```

Comment:
Looks like you are referring to a response here. While we only expect to stream binary **to** the server side. We would want a producer here for this. The global producer set for our swagger spec is `application/json` which should be good for our error model. Which is why this should function well with a defined `produces` section.